docs/Customization.md

```diff
@@ -0,0 +1,504 @@
+        $this->getModule('Doctrine2')->_reconfigure(['cleanup' => false]);
+    }
+
+{% endhighlight %}
+
+A group class can be created with `php vendor/bin/codecept generate:group groupname` command.
+Group classes will be stored in the `tests/_support/Group` directory.
+
+A group class can be enabled just like you enable an extension class. In the file `codeception.yml`:
+
+{% highlight yaml %}
+
+extensions:
+    enabled: [Group\Admin]
+
+{% endhighlight %}
+
+Now the Admin group class will listen for all events of tests that belong to the `admin` group.
+
+## Step Decorators
+
+Actor classes include generated steps taken from corresponding modules and helpers. 
+You can introduce wrappers for those steps by using step decorators. 
+
+Step decorators are used to implement conditional assertions. 
+When enabled, conditional assertions take all method prefixed by `see` or `dontSee` and introduce new steps prefixed with `canSee` and `cantSee`. 
+Contrary to standard assertions those assertions won't stop test on failure. This is done by wrapping action into try/catch blocks.
+
+List of available step decorators:
+
+- [ConditionalAssertion](https://github.com/Codeception/Codeception/blob/4.0/src/Codeception/Step/ConditionalAssertion.php)  - failed assertion will be logged, but test will continue.
+- [TryTo](https://github.com/Codeception/Codeception/blob/4.0/src/Codeception/Step/TryTo.php) - failed action will be ignored.
+- [Retry](https://github.com/Codeception/Codeception/blob/4.0/src/Codeception/Step/Retry.php) - failed action will be retried automatically.
+
+Step decorators can be added to suite config inside `steps` block:
+
+{% highlight yaml %}
+yml
+step_decorators:
+    - Codeception/Step/TryTo
+    - Codeception/Step/Retry
+    - Codeception/Step/ConditionalAssertion
+
+{% endhighlight %}
+
+You can introduce your own step decorators. Take a look into sample decorator classes and create your own class which implements `Codeception\Step\GeneratedStep` interface.
+A class should provide `getTemplate` method which returns a code block and variables passed into a template. 
+Make your class accessible by autoloader and you can have your own step decorators working. 
+
+## Custom Reporters
+
+Alternative reporters can be implemented as extension.
+There are [DotReporter](https://codeception.com/extensions#DotReporter) and [SimpleReporter](https://codeception.com/extensions#SimpleReporter) extensions included.
+Use them to change output or use them as an example to build your own reporter. They can be easily enabled with `--ext` option
+
+{% highlight bash %}
+
+php vendor/bin/codecept run --ext DotReporter
+
+{% endhighlight %}
+
+![](https://cloud.githubusercontent.com/assets/220264/26132800/4d23f336-3aab-11e7-81ba-2896a4c623d2.png)
+
+If you want to use it as default reporter enable it in `codeception.yml`.
+
+But what if you need to change the output format of the XML or JSON results triggered with the `--xml` or `--json` options?
```

Comment:
Text about custom PHPUnit printers must be removed because they are no longer supported in Codeception 5.

Reply:
oh, this change will break PHPStorm integration, because they use `-o "reporters: report: PhpStorm_Codeception_ReportPrinter"`

Reply:
Somebody raised an issue already: https://youtrack.jetbrains.com/issue/WI-65867/Codeception-5-does-not-support-reporters-option-anymore